switchapi/es2k/lnw_v3/switch_pd_lag.c

```diff
@@ -0,0 +1,872 @@
+/*
+ * Copyright 2023-2024 Intel Corporation.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "switch_pd_lag.h"
+
+#include "switchapi/es2k/switch_pd_p4_name_mapping.h"
+#include "switchapi/es2k/switch_pd_utils.h"
+#include "switchapi/switch_internal.h"
+#include "switchapi/switch_lag.h"
+#include "switchapi/switch_tdi.h"
+#include "switchutils/switch_log.h"
+
+/**
+ * Routine Description:
+ *   @brief Program tx_lag_table in MEV-TS
+ *
+ * Arguments:
+ *   @param[in] device - device
+ *   @param[in] lag_info - LAG info
+ *   @param[in] entry_add - true for entry add, false
+ *                          for entry delete
+ *
+ * Return Values:
+ *    @return  TDI_SUCCESS on success
+ *             Failure status code on error
+ */
+switch_status_t switch_pd_tx_lag_table_entry(switch_device_t device,
+                                             const switch_lag_info_t* lag_info,
+                                             bool entry_add) {
+  tdi_status_t status;
+
+  tdi_id_t field_id_lag_id = 0;
+  tdi_id_t field_id_hash = 0;
+  tdi_id_t action_id = 0;
+  tdi_id_t data_field_id = 0;
+
+  tdi_dev_id_t dev_id = device;
+
+  tdi_flags_hdl* flags_hdl = NULL;
+  tdi_target_hdl* target_hdl = NULL;
+  const tdi_device_hdl* dev_hdl = NULL;
```

Comment:
Why do we declare some of these variables as `const` and initialize to NULL? I see that this is then assigned in line 82

Reply:
pointer will be constant but data at that location will be not